Storage/Processor/Config.php

```diff
@@ -40,17 +42,19 @@ public function __construct(StorageManager $storageManager, string $processor, s
         $this->options = $this->resolve($options);
         $this->configHash = $configHash;
         $this->filename = null;
+        $this->cacheKey = $this->assetHash . '_' . $this->configHash;
 
-        if (!empty($this->getFileNames())) {
+        if ($this->getFileNames() !== null) {
```

Comment:
Did you not want to create a private function to assign the cacheKey property?

You could prevent having 3 nesting levles (if, foreach, if) by doing so !

Reply:
Created a setCacheKeyAndFilename function